### PR TITLE
Catalog my things

### PR DESCRIPTION
**Pull request by Timbar09**
In this PR, we have a console app that allows users to keep track of their possessions.

**Team**
- @Timbar09 - Team member NO.1
- @Tobby8629  - Team member NO.2
- @aubinleyoung  - Team member NO.3

**Changes**

- Created classes based on the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) in separate `.rb` files.
- Created the `Item` class which include the  `add_item` methods.
- Added child classes that inherit from `item`, the parent class.
- Added unit tests for all implemented methods.
- Created the console app entry point, `main.rb` which runs and displays the  User interface of the app.
- Preserved all data by saving collections in JSON files.
- Created a `schema.sql` file with tables that are analogical to the structure of the classes in the app.
- Included a link to the demo video of the app in the `README.md` project description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,8 +24,7 @@
 
 # 📖 Catalog <a name="about-project"></a>
 
-**Catalog** This project is a console app that will help me to keep a record of different types of things owned: books, music albums, movies, and games.. To to start up the App in the command prompt, run the following command [Ruby main.rb]
-
+> This project is a console application that allows the user to keep track of their possessions. It allows the user to manage various types of items such as books, music albums, and games. Here is a link to the video describing the application [Click](https://drive.google.com/drive/u/0/folders/1XwY5BwALptbISD5PAGEVwMO30yw9oK9l)
```
Comment (gemmen29) [inline README.md]: - [ ] Kindly make your video link public so I can view it.

File: classes/item.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  def initialize(id, publish_date, archived: false)
+    @id = id || Random.rand(1..1000)
+    @publish_date = publish_date
+    @archived = archived
+  end
+
+  def add_genre(genre)
+    @genre = genre
+  end
```
Comment (gemmen29) [inline classes/item.rb:14]: - [ ] Kindly add the `self` to the items list as you did in the `add_author` and `add_label` methods.

File: classes/app.rb [hunk 3]
```diff
@@ -0,0 +1,163 @@
+# require_relative 'book'
+# require_relative 'label'
+# require_relative 'author'
+# require_relative 'preserve_data'
```
Comment (gemmen29) [inline classes/app.rb:4]: - [ ] Please remove the commented code in order to make your repo clean.

File: classes/music.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# rubocop:disable Metrics/ParameterLists
```
Comment (gemmen29) [inline classes/music.rb:1]: - [ ] Kindly remove the Rubocop disable and fix the error manually as this defies the purpose of using linters in the first place.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Tobby8629** [comment]
@gemmen29  Thank you for the proper code review, it was helpful. I have carefully made changes to the projects according to the review made

**Review by Danieldotcomcoder — APPROVED** [review]
Hello Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Timbar09** [comment]
Thank you, @gemmen29 and @Danieldotcomcoder for reviewing and approving our code. 🙏

**Pull request merged**